trinity/sync/full/chain.py

```diff
@@ -85,314 +143,194 @@ class FastChainSyncer(BaseHeaderChainSyncer):
+        # store bodies for later usage, during block import
+        pending_bodies = {
+            header: bodies_by_root[root]
+            for header, root in completed_header_roots.items()
+        }
+        self._pending_bodies = merge(self._pending_bodies, pending_bodies)
+
+        self.logger.debug(
+            "Got block bodies for %d/%d headers from %s, from %r..%r",
+            len(completed_header_roots),
+            len(headers),
+            peer,
+            headers[0],
+            headers[-1],
+        )
 
-        self.logger.debug("Got receipts batch for %d headers", len(all_headers))
+        return block_body_bundles, completed_headers
 
-    async def _get_receipts(self,
-                            peer: ETHPeer,
-                            batch: Tuple[BlockHeader, ...],
-                            ) -> Tuple[Tuple[ReceiptBundle, ...], Tuple[BlockHeader, ...]]:
+    async def _request_block_bodies(
+            self,
+            peer: ETHPeer,
+            batch: Tuple[BlockHeader, ...]) -> Tuple[BlockBodyBundle, ...]:
         """
-        Requests the batch of receipts from the given peer, returning the
-        returned receipt data and the headers for which receipts were not
-        returned for.
+        Requests the batch of block bodies from the given peer, returning the
+        returned block bodies data, or an empty tuple on an error.
         """
-        self.logger.debug("Requesting receipts for %d headers from %s", len(batch), peer)
+        self.logger.debug("Requesting block bodies for %d headers from %s", len(batch), peer)
         try:
-            receipt_bundles = await peer.requests.get_receipts(batch)
+            block_body_bundles = await peer.requests.get_block_bodies(batch)
         except TimeoutError as err:
             self.logger.debug(
-                "Timed out requesting receipts for %d headers from %s", len(batch), peer,
+                "Timed out requesting block bodies for %d headers from %s", len(batch), peer,
             )
-            return tuple(), batch
+            return tuple()
         except CancelledError:
-            self.logger.debug("Pending receipts call to %r future cancelled", peer)
-            return tuple(), batch
+            self.logger.debug("Pending block bodies call to %r future cancelled", peer)
+            return tuple()
         except OperationCancelled:
-            self.logger.trace("Pending receipts call to %r operation cancelled", peer)
-            return tuple(), batch
+            self.logger.trace("Pending block bodies call to %r operation cancelled", peer)
+            return tuple()
         except PeerConnectionLost:
-            self.logger.debug("Peer went away, cancelling the receipts request and moving on...")
-            return tuple(), batch
+            self.logger.debug("Peer went away, cancelling the block body request and moving on...")
+            return tuple()
         except Exception:
```

Comment:
I guess we were already doing this but whatcha think about removing the bare `Exception` catching?  At this stage in the software I'd rather the node crash hard so that we're forced to address anything that bubbles out of there.